demosplan/DemosPlanCoreBundle/Command/Data/DeleteOrgaCommand.php

```diff
@@ -0,0 +1,341 @@
+            // delete all departments
+            $this->output->writeln('Deleting All Departments');
+            $this->deleteOrgaDepartments();
+            $this->processAllDepartments();
+            // delete all users
+            $this->output->writeln('Deleting All Users');
+            $this->deleteOrgaUsers();
+            $this->processAllUsers();
+            // delete all procedures
+            $this->output->writeln('Deleting All Procedures');
+            $this->deleteProcedures();
+            // delete all Settings
+            $this->output->writeln('Deleting All Settings');
+            $this->deleteSettings();
+            // delete all StatementMeta
+            $this->output->writeln('Deleting All StatementMeta');
+            $this->deleteStatementMeta();
+            // delete all Slug
+            $this->output->writeln('Deleting All Slug');
+            $this->deleteOrgaSlug();
+            $this->deleteSlug();
+            // delete orga
+            $this->output->writeln('Deleting Orga');
+            $this->deleteOrga();
+            // reactivate foreign key checks
+            $this->output->writeln('Activate FK Checks');
+            $this->activateForeignKeyChecks();
+
+            // commit all changes
+            $this->output->writeln('Committing all changes');
+            $this->dbConnection->commit();
+
+            // repopulate Elasticsearch
+            $this->repopulateElasticsearch();
+
+            $this->output->writeln("Procedure $this->orgaId was purged successfully!");
+
+            return Command::SUCCESS;
+        } catch (Exception $e) {
+            // rollback all changes
+            $this->dbConnection->rollBack();
+            $this->output->writeln('Rolled back transaction');
+
+            $this->output->error($e->getMessage());
+            $this->output->error($e->getTraceAsString());
+
+            return Command::FAILURE;
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function deleteAddress(array $addressesID): void
+    {
+        $this->deleteFromTableByIdentifierArray('_address', '_a_id', $addressesID);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function deleteOrgaAddressess(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_orga_addresses_doctrine', '_o_id', [$this->orgaId]);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function deleteDepartments(array $departmentsId): void
+    {
+        $this->deleteFromTableByIdentifierArray('_department', '_d_id', $departmentsId);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function deleteOrgaDepartments(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_orga_departments_doctrine', '_o_id', [$this->orgaId]);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function deleteUsers(array $usersId): void
+    {
+        $this->deleteFromTableByIdentifierArray('_user', '_u_id', $usersId);
```

Comment:
Afaik your system can not handle deleted users at all.

Reply:
Additionally this and also the other relations of an orga should be delete correctly by doctrine via cascade deletion. If this does not happend yet, why not fixing this instead of creating new logic?